core/tests/test_views.py

```diff
@@ -0,0 +1,659 @@
+        assert response_data["name"] == updated_data["name"]
+        assert response_data["is_for_adults"] == updated_data["is_for_adults"]
+
+    def test_genre_delete(self, client):
+        genre = GenreFactory.create()
+        url = reverse("genre-detail", args=[genre.id])
+
+        response = client.delete(url)
+
+        assert response.status_code == 204
+        assert not Genre.objects.filter(id=genre.id).exists()
+
+
+@pytest.mark.django_db
+class TestActor:
+    url_list = reverse("actor-list")
+
+    def test_actor_post(self, client):
+        actor = ActorFactory.build()
+        serialized = ActorSerializer(actor).data
+
+        response = client.post(
+            self.url_list, data=serialized, content_type="application/json"
+        )
+        response_data = response.json()
+
+        assert response.status_code == 201
+        assert response_data["name"] == serialized["name"]
+        assert response_data["age"] == serialized["age"]
+        assert response_data["nationality"] == serialized["nationality"]
+        assert Actor.objects.filter(name=serialized["name"]).exists()
+
+    def test_actor_post_blank(self, client):
+        expected = {
+            "name": ["This field is required."],
+            "age": ["This field is required."],
+            "nationality": ["This field is required."],
+        }
+
+        response = client.post(self.url_list, data={}, content_type="application/json")
+
+        assert response.status_code == 400
+        assert response.json() == expected
+
+    def test_actor_post_invalid_name(self, client):
+        data = {
+            "name": "a" * 81,
+            "age": 30,
+            "nationality": "TestCountry",
+        }
+
+        response = client.post(
+            self.url_list, data=data, content_type="application/json"
+        )
+
+        assert response.status_code == 400
+        assert response.json() == {
+            "name": ["Ensure this field has no more than 80 characters."]
+        }
+
+    def test_actor_get_list(self, client):
+        ActorFactory.create()
+        ActorFactory.create()
+        expected_keys = {"id", "name", "age", "nationality"}
+
+        response = client.get(self.url_list)
+        response_data = response.json()
+
+        assert response.status_code == 200
+        assert len(response_data) == 2
+        assert expected_keys == set(response_data[0].keys())
+
+    def test_actor_get_detail(self, client):
+        actor = ActorFactory.create()
+        url = reverse("actor-detail", args=[actor.id])
+
+        response = client.get(url)
+        response_data = response.json()
+
+        assert response.status_code == 200
+        assert response_data["name"] == actor.name
+
+    def test_actor_put(self, client):
+        actor = ActorFactory.create()
+        url = reverse("actor-detail", args=[actor.id])
+        new_data = {
+            "name": "Updated Actor",
+            "age": actor.age + 1,
```

Comment:
maybe nitpicking maybe not. If we can avoid any state if test I would do that. If we can put fixed age into factory I would do that, so test always tests the same. Also this test might fail with validation error, if for some reason factory picks highest legal age value, and test tries updating it to +1